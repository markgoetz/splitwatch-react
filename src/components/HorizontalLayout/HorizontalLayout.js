import React from 'react';
import PropTypes from 'prop-types';
import styles from './HorizontalLayout.module.css';

export default function HorizontalLayout(props) {
    let additionalClassname = '';
  
    switch (props.align) {
        case 'top':
            additionalClassname = styles.top;
            break;
        case 'middle':
            additionalClassname = styles.middle;
            break;
        case 'bottom':
            additionalClassname = styles.bottom;
            break;
        default:
            break;
    }

    const className = styles.horizontalLayout + ' ' + additionalClassname;

    return (<div className={className}>
        {props.children}
    </div>);
}

HorizontalLayout.propTypes = {
    align: PropTypes.oneOf(['top', 'middle', 'bottom'])
};