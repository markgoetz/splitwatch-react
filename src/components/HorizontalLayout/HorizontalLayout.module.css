.horizontalLayout {
  display: grid;
  grid-gap: 10px;
  grid-auto-flow: column;
}

@media (min-width: 769px) {
  .horizontalLayout {
    grid-gap: 20px;
  }
}

.top {
  align-items: flex-start;
}

.middle {
  align-items: center;
}

.bottom {
  align-items: flex-end;
}