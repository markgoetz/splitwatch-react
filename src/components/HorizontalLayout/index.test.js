import React from 'react';
import HorizontalLayout from './HorizontalLayout';
import { shallow } from 'preact-render-spy';

describe('Horizontal Layout', () => {

  test('Horizontal layout renders correctly', () => {
    const context = shallow(<HorizontalLayout>
      <div>Test element</div>
      <button>Test button</button>
      <div>Extra test element</div>
    </HorizontalLayout>);
    expect(context).toMatchSnapshot();
  });

  test('Horizontal layout renders correctly when top-aligned', () => {
    const context = shallow(<HorizontalLayout align="top">
      <div>Test element</div>
      <button>Test button</button>
      <div>Extra test element</div>
    </HorizontalLayout>);
    expect(context).toMatchSnapshot();
  });

  test('Horizontal layout renders correctly when centered', () => {
    const context = shallow(<HorizontalLayout align="middle">
      <div>Test element</div>
      <button>Test button</button>
      <div>Extra test element</div>
    </HorizontalLayout>);
    expect(context).toMatchSnapshot();
  });

  test('Horizontal layout renders correctly when bottom-aligned', () => {
    const context = shallow(<HorizontalLayout align="bottom">
      <div>Test element</div>
      <button>Test button</button>
      <div>Extra test element</div>
    </HorizontalLayout>);
    expect(context).toMatchSnapshot();
  });
});