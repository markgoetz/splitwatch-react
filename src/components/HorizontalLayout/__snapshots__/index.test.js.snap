// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Horizontal Layout Horizontal layout renders correctly 1`] = `
preact-render-spy (1 nodes)
-------
<div class="undefined ">
  <div>Test element</div>
  <button>Test button</button>
  <div>Extra test element</div>
</div>

`;

exports[`Horizontal Layout Horizontal layout renders correctly when bottom-aligned 1`] = `
preact-render-spy (1 nodes)
-------
<div class="undefined undefined">
  <div>Test element</div>
  <button>Test button</button>
  <div>Extra test element</div>
</div>

`;

exports[`Horizontal Layout Horizontal layout renders correctly when centered 1`] = `
preact-render-spy (1 nodes)
-------
<div class="undefined undefined">
  <div>Test element</div>
  <button>Test button</button>
  <div>Extra test element</div>
</div>

`;

exports[`Horizontal Layout Horizontal layout renders correctly when top-aligned 1`] = `
preact-render-spy (1 nodes)
-------
<div class="undefined undefined">
  <div>Test element</div>
  <button>Test button</button>
  <div>Extra test element</div>
</div>

`;
